src/google/protobuf/compiler/cpp/cpp_helpers.h

```diff
@@ -746,21 +746,21 @@ class PROTOC_EXPORT NamespaceOpener {
   void ChangeTo(const std::string& name) {
     std::vector<std::string> new_stack_ =
         Split(name, "::", true);
-    int len = std::min(name_stack_.size(), new_stack_.size());
-    int common_idx = 0;
+    size_t len = std::min(name_stack_.size(), new_stack_.size());
+    size_t common_idx = 0;
     while (common_idx < len) {
       if (name_stack_[common_idx] != new_stack_[common_idx]) break;
       common_idx++;
     }
-    for (int i = name_stack_.size() - 1; i >= common_idx; i--) {
-      if (name_stack_[i] == "PROTOBUF_NAMESPACE_ID") {
+    for (auto oldNameIt = name_stack_.crbegin(); oldNameIt != name_stack_.crend() - common_idx; ++oldNameIt) {
```

Comment:
Nice, this looks like a good solution. One more small thing, though: `oldNameIt` should be in `snake_case`, maybe rename it to `old_name_iter` or even just `iter`?

Reply:
I have changed it to simple `it`, which is common in the code I write. I hope such naming will fit.